From a3a2e5e239318af2b4660db65ad493ce305384de Mon Sep 17 00:00:00 2001
From: Jack Tan <jack.tan@windriver.com>
Date: Wed, 12 Feb 2014 17:31:32 +0800
Subject: [PATCH 10/58] bcm-xlp2: numa support

XLP NUMA support. The key init function 'xlp_numa_bootmem_init' is defined
in the 'arch/mips/netlogic/xlp/numa.c'

Based on SDK 3.0 (2013-10-29)

Signed-off-by: Jack Tan <jack.tan@windriver.com>
---
 arch/mips/kernel/setup.c |    9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/arch/mips/kernel/setup.c b/arch/mips/kernel/setup.c
index c7f9051..6b8b0be 100644
--- a/arch/mips/kernel/setup.c
+++ b/arch/mips/kernel/setup.c
@@ -290,6 +290,15 @@ static void __init bootmem_init(void)
 	finalize_initrd();
 }
 
+#elif defined(CONFIG_CPU_XLP) && defined(CONFIG_NUMA)
+extern void xlp_numa_bootmem_init(void);
+static void __init bootmem_init(void)
+{
+	init_initrd();
+	xlp_numa_bootmem_init();
+	finalize_initrd();
+}
+
 #else  /* !CONFIG_SGI_IP27 */
 
 static void __init bootmem_init(void)
-- 
1.7.9.5

